src/PSDParametrizations/PSDParametrizations.jl

```diff
@@ -0,0 +1,29 @@
+
+abstract type PSDTrait end
+struct IsPSDParametrization <: PSDTrait end
+struct IsNotPSDParametrization <: PSDTrait end
+
+ispsdparametrization(::Any) = IsNotPSDParametrization()
```

Comment:
Is it important that it is a "parameterization", or is the defining thing here that the object is PSD? In the latter case, I think just having `is_psd` and `IsPSD`/`IsNotPSD`, or even `PSD/NotPSD` might also be fine (if we don't go the `True`/`False` route)

Reply:
Let's narrow the discussion to matrices (rather than general operators) for now. The property we are trying to discern here is whether a specific type can conceivably be thought to represent a positive-semi definite matrix rather than anything else. This as opposed to discerning whether a certain type representing a matrix is PSD or not. I feel like ```IsPSD``` / ```IsNotPSD``` could be interpreted as the latter, which would be misleading. 

In that sense, I guess a more accurate naming would  be: 

```julia
struct CouldBePSD <: PSDTrait end 
struct CouldNotBePSD <: PSDTrait end 
```

On the other hand, 'could be PSD' would apply to any subtype of ```AbstractMatrix``` so could be interpreted as more wide than I intend. On the other hand, any subtype ```AbstractMatrix``` need not be square so from the type level it would only apply to ```Symmetric``` and ```Hermitian``` so maybe fine.

In any case, I would be open to more intuitive namings of this concept.